### PR TITLE
feat(amzonq): loosen inline suggestion restriction for json/yaml files

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Problem

JB PR https://github.com/aws/aws-toolkit-jetbrains/pull/4928

Per product team's request, change condition when we support generateCompletion for YAML/JSON

YAML files

- Provide completions for all YAML files.

- Remove any current CloudFormation-only restrictions for YAML

JSON files

- Provide completions for a JSON file if the filename matches one of the below names or the file content matches the CloudFormation regexp.
```
    app.json
    appsettings.json
    bower.json
    composer.json
    db.json
    manifest.json
    package.json
    schema.json
    settings.json
    tsconfig.json
    vcpkg.json
```

## Solution

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/util/commonUtil.ts [hunk 1]
```diff
@@ -61,11 +65,12 @@ export function getPrefixSuffixOverlap(firstString: string, secondString: string
...
 
-export function checkLeftContextKeywordsForJsonAndYaml(leftFileContent: string, language: string): boolean {
+export function checkKeywordsForJson(fileName: string, leftFileContent: string, language: string): boolean {
     if (
-        (language === 'json' || language === 'yaml') &&
+        language === 'json' &&
         !AWSTemplateKeyWords.some((substring) => leftFileContent.includes(substring)) &&
-        !AWSTemplateCaseInsensitiveKeyWords.some((substring) => leftFileContent.toLowerCase().includes(substring))
+        !AWSTemplateCaseInsensitiveKeyWords.some((substring) => leftFileContent.toLowerCase().includes(substring)) &&
+        !JsonConfigFileNamingConvention.has(fileName)
```
Comment (andrewyuq) [inline packages/core/src/codewhisperer/util/commonUtil.ts:73]: do we want to allow filenames ignore cases or no

Comment (Will-ShaoHua) [reply packages/core/src/codewhisperer/util/commonUtil.ts:73]: that's a good callout, i think better to have case ignore.

**Review by zixlin7 — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Review by nkomonen-amazon — APPROVED** [review]

**Comment by Will-ShaoHua** [comment]
2631 passing (1m)
  12 pending
  1 failing
  1) startSecurityScanPerformanceTest
       Should calculate cpu and memory usage for file scans performance tests
         "after all" hook for "Should calculate cpu and memory usage for file scans - test run 10":

      AssertionError [ERR_ASSERTION]: Expected total user CPU usage for Should calculate cpu and memory usage for file scans to be less than 50. Actual user CPU usage was 50.67705300541362
      + expected - actual

      -false
      +true
      
      at assertPerformanceMetrics (/codebuild/output/src502230826/src/github.com/aws/aws-toolkit-vscode/packages/core/src/shared/performance/performance.ts:182:11)
      at Context.<anonymous> (/codebuild/output/src502230826/src/github.com/aws/aws-toolkit-vscode/packages/core/src/shared/performance/performance.ts:160:13)
      at processImmediate (node:internal/timers:476:21)
      at process.topLevelDomainCallback (node:domain:161:15)
      at process.callbackTrampoline (node:internal/async_hooks:128:24)

**Comment by Will-ShaoHua** [comment]
2631 passing (1m)
  12 pending
  1 failing
  1) startSecurityScanPerformanceTest
       Should calculate cpu and memory usage for file scans performance tests
         "after all" hook for "Should calculate cpu and memory usage for file scans - test run 10":

      AssertionError [ERR_ASSERTION]: Expected total user CPU usage for Should calculate cpu and memory usage for file scans to be less than 50. Actual user CPU usage was 50.67705300541362
      + expected - actual

      -false
      +true
      
      at assertPerformanceMetrics (/codebuild/output/src502230826/src/github.com/aws/aws-toolkit-vscode/packages/core/src/shared/performance/performance.ts:182:11)
      at Context.<anonymous> (/codebuild/output/src502230826/src/github.com/aws/aws-toolkit-vscode/packages/core/src/shared/performance/performance.ts:160:13)
      at processImmediate (node:internal/timers:476:21)
      at process.topLevelDomainCallback (node:domain:161:15)
      at process.callbackTrampoline (node:internal/async_hooks:128:24)

File: packages/amazonq/.changes/next-release/Feature-88ac445a-7a5a-4127-97f7-eaccd41629e6.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "Loose inline completion support limitation for yaml/json files"
```
Comment (justinmk3) [inline packages/amazonq/.changes/next-release/Feature-88ac445a-7a5a-4127-97f7-eaccd41629e6.json:3]: Thanks for including a changelog. Please review the [changelog guidelines](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#changelog).

Your PR description gives a hint about the actual, customer-facing, behavior change:


```suggestion
	"description": "Inline completion for more json files, and all yaml files"
```

**Pull request merged**